speed-dreams/source-2.2.3/src/libs/tgfdata/tracks.cpp

```diff
@@ -529,6 +529,33 @@ int GfTrack::getMaxNumOfPitSlots() const
 	return _nMaxPitSlots;
 }
 
+/// SIMULATED DRIVING ASSISTANCE: added the getEstimatedTime function
+/// @brief  Gets the estimated time it takes to complete one round of a track
+/// @return The estimated time it takes to complete one round of a track
+double GfTrack::GetEstimatedTime() const 
+{
+   // Lazy load scheme : read files only when really needed, and only once.
+    if (_fEstimatedTime < 0)
+        load();
+    return _fEstimatedTime;
+}
+
+/// SIMULATED DRIVING ASSISTANCE: added getSpeedLimit function
+/// @brief  Gets the speed limit of the track
+/// @return The speed limit of the track which is either 80 km/h or 100 km/h
+double GfTrack::GetSpeedLimit() const 
+{
+    const std::string& description = getDescription();
+
+	// If there is an 8 in the string, the speed limit is 80 km/h
+	// otherwise the speed limit is 100 km/h
+	char c = '8';
+    if (description.find(c) == c)
+        return 80;
+    else
+        return 100;
+}
```

Comment:
This is unmaintainable code:
- First of all it assumes that the speed is mentioned in the description, which is not neccessarily the case.
- Second of all it will need an update if someone ever wants to use a different speed limit.
- This currently always returns 100 because it uses the `find` functions return type incorrectly.

Either just add an attribute to the tracks with a hardcoded estimated time, or wait for Sjoerds PR, which adds speed limits for every segment of a track. Then you can use these values to calculate the 'estimated time' of a track.

Reply:
I fully agree with all your points, I will communicate with Sjoerd and adjust the code according to his PR so that the code can be improved.

Reply:
I believe that the code is enough maintainable now.